### PR TITLE
Move GceMetadata fetching to start_esp.

### PR DESCRIPTION
**Pull request by qiwzhang**
The current way is using Nginx custom upstream http.cc to fetch "computeMetadata/v1/?recursive=true"
whose response is too big. Sometime it causes "connection reset by peer" error which the custom http upstream does 't handle correctly, results in Nginx crash.  

ESP only needs 3, 4 fields.   move such fetch to start_esp before Nginx starts.  Use server_config to pass the result into ESP code.

### CODE REVIEW

File: src/api_manager/auth/service_account_token.h [hunk 1]
```diff
@@ -54,6 +54,14 @@ class ServiceAccountToken {
     access_token_.set_token(token, expiration);
   }
 
+  // The access token from metadata server response inn JSON format:
```
Comment (JimmyCYJ) [inline src/api_manager/auth/service_account_token.h]: inn->in

File: src/api_manager/proto/server_config.proto [hunk 2]
```diff
@@ -247,3 +250,25 @@ message Experimental {
...
+  // The value from "project/project-id"
+  string project_id = 1;
+  // The value from "instance/zone"
+  string zone = 2;
+  // The value from "instance/attributes/gae_server_software"
+  string gae_server_software = 3;
+  // The value from "instance/attributes/kube-env"
+  string kube_env = 4;
+  // The access token from default service account
+  AccessToken access_token = 5;
```
[the remark this answers is not in the document]
Comment (qiwzhang) [reply src/api_manager/proto/server_config.proto]: Yes.  But start_esp will get the first one,  and ESP will re-fresh it when it expires.  This change will save the first fetch by ESP.

**Pull request merged**